src/middleware/messageStore.coffee

```diff
@@ -130,6 +100,56 @@ exports.storeResponse = (ctx, done) ->
       return done err
     return done()
 
+exports.storeNonPrimaryResponse = (ctx, routeObject, done) ->
+  # check if channel response body is false and remove
+  if ctx.authorisedChannel.responseBody == false
+    response.response.body = ''
+
+  if ctx.transactionId?
```

Comment:
Perhaps add an else statement here and log and error if the transactionId doesn't exist.